File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetDisplayPlaneSupportedDisplaysKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetDisplayPlaneSupportedDisplaysKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jint planeIndex, jlong pDisplayCount, jlong pDisplays){
    (void)env;
    (void)jc;
    PFN_vkGetDisplayPlaneSupportedDisplaysKHR f = (PFN_vkGetDisplayPlaneSupportedDisplaysKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        (uint32_t)planeIndex,
        ((uint32_t*)jniLongToPointer(pDisplayCount)),
        ((VkDisplayKHR*)jniLongToPointer(pDisplays))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkGetDisplayPlaneSupportedDisplaysKHR.java
package cz.mg.vulkan;

public class PFNvkGetDisplayPlaneSupportedDisplaysKHR extends VkFunctionPointer {
    public PFNvkGetDisplayPlaneSupportedDisplaysKHR() {
    }

    protected PFNvkGetDisplayPlaneSupportedDisplaysKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetDisplayPlaneSupportedDisplaysKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDisplayPlaneSupportedDisplaysKHR(long value) {
        setValue(value);
    }

    public PFNvkGetDisplayPlaneSupportedDisplaysKHR(VkInstance instance) {
        super(instance, new VkString("vkGetDisplayPlaneSupportedDisplaysKHR"));
    }

    public void call(VkPhysicalDevice physicalDevice, VkUInt32 planeIndex, VkUInt32 pDisplayCount, VkDisplayKHR pDisplays, VkResult rval){
        callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), planeIndex != null ? planeIndex.getVkAddress() : VkPointer.getNullAddressNative(), pDisplayCount != null ? pDisplayCount.getVkAddress() : VkPointer.NULL, pDisplays != null ? pDisplays.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long physicalDevice, long planeIndex, long pDisplayCount, long pDisplays, long rval);


    public int call(VkPhysicalDevice physicalDevice, int planeIndex, VkUInt32 pDisplayCount, VkDisplayKHR pDisplays){
        return callSimplifiedNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), planeIndex, pDisplayCount != null ? pDisplayCount.getVkAddress() : VkPointer.NULL, pDisplays != null ? pDisplays.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long physicalDevice, int planeIndex, long pDisplayCount, long pDisplays);

}
